include/maths/matrices/Matrix4.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef MATRIX4_HPP
+#define MATRIX4_HPP
+#include <array>
+#include <Vector4.hpp>
+
+class Matrix4
+{
+public:
+    // Constructors
+    Matrix4() = default;
+    explicit Matrix4(const std::array<float, 16>&);
+    Matrix4(const Matrix4& other) = default;
+    
+    // Destructor
+    ~Matrix4() = default;
+    
+    // Getters
+    [[nodiscard]] const float* getData() const;
+    [[nodiscard]] Vector4 getRowOne() const;
+    [[nodiscard]] Vector4 getRowTwo() const;
+    [[nodiscard]] Vector4 getRowThree() const;
+    [[nodiscard]] Vector4 getRowFour() const;
+    [[nodiscard]] Vector4 getColumnOne() const;
+    [[nodiscard]] Vector4 getColumnTwo() const;
+    [[nodiscard]] Vector4 getColumnThree() const;
+    [[nodiscard]] Vector4 getColumnFour() const;
```

Comment:
T'aurais pas mieux fait de faire ca ?
```suggestion
    [[nodiscard]] Vector4 getRow(int rowIndex) const;
    [[nodiscard]] Vector4 getColumn(int columnIndex) const;
```

Reply:
Fait